src/Disks/IVolume.cpp

```diff
@@ -49,9 +49,9 @@ IVolume::IVolume(
         throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "Volume must contain at least one disk");
 }
 
-UInt64 IVolume::getMaxUnreservedFreeSpace() const
+std::optional<UInt64> IVolume::getMaxUnreservedFreeSpace() const
 {
-    UInt64 res = 0;
+    std::optional<UInt64> res = 0;
```

Comment:
I think it should be uninitialized to yield nullopt value in case of all the disks have nullopt
```suggestion
    std::optional<UInt64> res;
```